client/post-editor/controller.js

```diff
@@ -247,7 +247,12 @@ export default {
 				startEditingPostCopy( site, postToCopyId, context );
 			} else if ( postID ) {
 				// TODO: REDUX - remove flux actions when whole post-editor is reduxified
-				actions.startEditingExisting( site, postID );
+				actions.startEditingExisting( site, postID ).then( editedPost => {
+					if ( editedPost && editedPost.type && editedPost.type !== postType ) {
```

Comment:
what happens if this api call is slow and the user leaves the editor before it returns?

Reply:
That's a good point! You'll be navigated back to the editor against your will. I pushed a fix that checks if `page` is still handling the same route. I tested it by adding a few seconds timeout into the promise chain: if I stay in the editor, I get redirected, if I close it, I don't.

Reply:
woo, nice!